File: lib/java/src/main/java/uapi/internal/validation/MapValidationFailuresToInvalidFieldCases.java
package uapi.internal.validation;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class MapValidationFailuresToInvalidFieldCases {

    public static List<Map<String, Object>> mapValidationFailuresToInvalidFieldCases(
            List<ValidationFailure> argumentValidationFailures) {
        final var validationFailureCases = new ArrayList<Map<String, Object>>();
        for (final var validationFailure : argumentValidationFailures) {
            final Map<String, Object> validationFailureCase = Map.of(
                    "path", validationFailure.path,
                    "reason", Map.of(validationFailure.reason, validationFailure.data));
            validationFailureCases.add(validationFailureCase);
        }

        return validationFailureCases;
    }
}

File: lib/ts/src/internal/validation/MapValidationFailuresToInvalidFieldCases.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { ValidationFailure } from '../../internal/validation/ValidationFailure.js';

export function mapValidationFailuresToInvalidFieldCases(
    argumentValidationFailures: ValidationFailure[],
): Record<string, any>[] {
    const validationFailureCases: Record<string, any>[] = [];
    for (const validationFailure of argumentValidationFailures) {
        const validationFailureCase = {
            path: validationFailure.path,
            reason: { [validationFailure.reason]: validationFailure.data },
        };
        validationFailureCases.push(validationFailureCase);
    }

    return validationFailureCases;
}
